enhancements/sig-architecture/6-placements/README.md

```diff
@@ -0,0 +1,523 @@
+// ManagedCluster should be evicted accordingly.
+type Placement struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the attributes of Placement.
+	Spec PlacementSpec `json:"spec"`
+
+	// Status represents the current status of the Placement
+	// +optional
+	Status PlacementStatus `json:"status,omitempty"`
+}
+
+// PlacementSpec defines the attributes of Placement.
+// An empty PlacementSpec selects all clusters from the bound ManagedClusterSets to placement namespace.
+type PlacementSpec struct {
+	// Predicates represent a slice of predicates to select clusters. The predicates are ORed.
+	// +optional
+	Predicates []ClusterPredicate `json:"predicates,omitempty"`
+
+	// Affinity represents the scheduling constraints
+	// +optional
+	Affinity *Affinity `json:"affinity,omitempty"`
+}
+
+// ClusterPredicate represents a predicate to select clusters.
+type ClusterPredicate struct {
+	// ClusterSets represent the ManagedClusterSets from which the clusters are selected.
+	// If the slice is empty, clusters will be selected from the ManagedClusterSets bound to the placement
+	// namespace, otherwise clusters will be selected from the intersection of this slice and the
+	// ManagedClusterSets bound to the placement namespace.
+	// +optional
+	ClusterSets []string `json:"clusterSets,omitempty"`
+
+	// NumberOfClusters represents the number of clusters to be selected which meet the
+	// requirements in this predicate.
+	// Nil indicates that NumberOfClusters is not set and there is no bound number.
+	// +optional
+	NumberOfClusters *int32 `json:"numberOfClusters,omitempty"`
+
+	// RequiredDuringSchedulingRequiredDuringExecution represents a scheduling strategy for a
+	// cluster selector. If specified, the cluster selector should be matched at both scheduling
+	// time (when the placement decisions are made for the first time) and execution time (After
+	// one or multiple placement decisions are made). It means,
+	// 1) Any cluster, which does not match the selector, should not be selected;
+	// 2) If a selected cluster ceases to match the selector (e.g. due to an update) after the
+	//    placement decisions are made, it will be eventually removed from the placement decisions;
+	// 3) If a cluster starts to match the selector after the placement decisions are made, it
+	//    will either be selected or at least has chance to be selected (when NumberOfClusters is
+	//    specified);
+	//
+	// Other scheduling strategies not yet implemented includes
+	// RequiredDuringSchedulingIgnoredDuringExecution and PreferredDuringSchedulingIgnoredDuringExecution
+	// +optional
+	RequiredDuringSchedulingRequiredDuringExecution *ClusterSelector `json:"requiredDuringSchedulingRequiredDuringExecution,omitempty"`
+}
+
+// ClusterSelector represents the AND of the containing selectors
+type ClusterSelector struct {
+	// LabelSelector represents a selector of clusters by label
```

Comment:
> I might suggest a more brief form of this name `requiredDuringSchedulingAndExecution`

Regarding `RequiredDuringSchedulingRequiredDuringExecution`, we borrowed this field name from NodeAffinity. They have the similar meanings. So I think it might be better to keep this filed name, for users may have already been familiar with it in NodeAffinity.

Reply:
> Label on which resource exactly? I think it will pay to be specific here.

Updated the comments with the resource type name.